src/id-data-controls.js

```diff
@@ -192,58 +236,69 @@ class IdDataControls extends preact.Component {
                 desc: t('name', 'generator'),
                 type: 'name',
                 value: data['name'],
-                optional: true
+                optional: true,
             },
             {
                 desc: t('birthdate', 'generator'),
                 type: 'birthdate',
                 value: data['birthdate'],
-                optional: true
+                optional: true,
             },
             {
                 desc: t('address', 'generator'),
                 type: 'address',
                 value: data['address'],
-                optional: true
+                optional: true,
             },
             {
                 desc: t('email-address', 'generator'),
                 type: 'email',
                 value: data['email'],
-                optional: true
-            }
+                optional: true,
+            },
         ];
     }
 
     resetSavedIdData() {
-        this.savedIdData.getAll().then(id_data => this.setState({ custom_id_data: id_data }));
-        this.savedIdData.getAllFixed().then(fixed_data => {
-            this.setState(prev => {
+        this.savedIdData.getAll().then((id_data) => this.setState({ custom_id_data: id_data }));
+        this.savedIdData.getAllFixed().then((fixed_data) => {
+            this.setState((prev) => {
                 for (let i in fixed_data) {
                     prev.fixed_id_data[fixed_data[i].type] = fixed_data[i].value;
                 }
                 return prev;
             });
         });
-        this.savedIdData.getSignature().then(signature => this.setState({ signature: signature }));
+        this.savedIdData.getSignature().then((signature) => this.setState({ signature: signature }));
     }
 
-    componentDidMount() {
-        window.addEventListener(ID_DATA_CHANGE_EVENT, event => {
+    queued = false;
+    enqueueReset() {
+        if (this.queued) return;
+        this.queued = true;
+        Promise.resolve().then(() => {
             this.resetSavedIdData();
+            this.queued = false;
         });
-        window.addEventListener(ID_DATA_CLEAR_EVENT, event => {
-            this.resetSavedIdData();
+    }
+
+    componentDidMount() {
+        window.addEventListener(ID_DATA_CHANGE_EVENT, (event) => {
+            this.enqueueReset();
+        });
+        window.addEventListener(ID_DATA_CLEAR_EVENT, (event) => {
+            this.enqueueReset();
         });
     }
 }
```

Comment:
This is actually a really nice way to solve this. However, I think we no longer need the events anyway, since we retired the id data widget we had on top of the page. But now that you implemented it…